packages/synapse-react-client/src/components/CardContainer/CardContainer.tsx

```diff
@@ -61,13 +61,28 @@ export function CardContainer(props: CardContainerProps) {
     isLoading,
     secondaryLabelLimit = 3,
     title,
+    initialLimit,
     ...rest
   } = props
   const infiniteQueryContext = useInfiniteQueryContext()
   const { NoContentPlaceholder } = useQueryVisualizationContext()
   const { appendNextPageToResults, hasNextPage } = infiniteQueryContext
   const data = useAtomValue(tableQueryDataAtom)
   const queryVisualizationContext = useQueryVisualizationContext()
+  const [hasAppliedInitialLimit, setHasAppliedInitialLimit] =
+    React.useState<boolean>(false)
+
+  const queryCount = data?.queryCount
+  const applyInitialLimit =
+    initialLimit !== undefined &&
+    queryCount !== undefined &&
+    initialLimit < queryCount &&
+    !hasAppliedInitialLimit
```

Comment:
Handle case where initialLimit is greater than the total number of results for this query, in which case we shouldn't change the number of cards shown initially